### DISCUSSION TITLE
Ribbons rendering differently when in a Multiblock

### DISCUSSION THREAD

**Discussion by josh-millar** [pyvista/pyvista#4897] [Q&A]
I am using pyvista to render ribbons from a geopandas dataframe. I convert each line to a ribbon, then append it a MultiBlock:

```
import osmnx as ox
import matplotlib.pyplot as plt
import numpy as np
import pyvista as pv
import geopandas as gpd

latlon = [xxx,yyy]
graph = ox.graph_from_point(latlon, dist=200, network_type='drive')
nodes, edges = ox.graph_to_gdfs(graph)
lineblock = pv.MultiBlock()

def lines_from_points(points):
	poly = pv.PolyData()
	poly.points = points
	this_cell = np.arange(0, len(points), dtype=np.int_)
	this_cell = np.insert(this_cell, 0, len(points))
	# cell_type = np.array([4])
	# this_grid = pv.UnstructuredGrid(this_cell, cell_type, points)
	poly.lines = this_cell

	return poly

for idx, row in edges.iterrows():
	x_pts = row['geometry'].xy[0]
	y_pts = row['geometry'].xy[1]
	z_pts = np.zeros(len(x_pts))
	pts = np.column_stack((x_pts, y_pts, z_pts))
	this_line = lines_from_points(pts)
	this_ribbon = this_line.ribbon(width=1e-5, normal=(0.0, 0.0, 1.0))
	# this_ribbon.save('./'+str(x_pts)+ 'ribbon.vtk')
	lineblock.append(this_ribbon)

combined_lines = lineblock.combine()
combined_lines.save('./test.vtk', binary=False)
```
However, the resulting output is not as expected:
![image](https://user-images.githubusercontent.com/89454053/130600860-3228956c-84d5-4cef-89ca-386ba338a1d2.png)
However, if I save each ribbon as a vtk, this works as expected:
![image](https://user-images.githubusercontent.com/89454053/130600770-182dc3ff-671d-4334-96c3-36c591b5a477.png)
So the issue is somewhere with the MultiBlock. 

Does anyone know why this might be happening and a possible workaraound?

**Comment by akaszynski** [comment]
That's indeed strange.  I'd have to see the full code (including the data to be able to reproduce this.

**Comment by josh-millar** [comment]
That is the whole code - osmnx handles the fetching of the dataset. You just need to change xxx,yyy with a valid lat/lon.

**Comment by akaszynski** [comment]
Sorry @josh-millar, glanced at this during work and didn't see that.

This is a really cool example!  The following is an alternative to using multi-blocks:

```py
import osmnx as ox
import numpy as np
import pyvista as pv


address = 'Holzgerlingen DE'
graph = ox.graph_from_address(address, dist=500, network_type='drive')

nodes, edges = ox.graph_to_gdfs(graph)
lines = []

# convert each edge into a line
for idx, row in edges.iterrows():
    x_pts = row['geometry'].xy[0]
    y_pts = row['geometry'].xy[1]
    z_pts = np.zeros(len(x_pts))
    pts = np.column_stack((x_pts, y_pts, z_pts))
    this_line = pv.lines_from_points(pts)
    lines.append(this_line)

# merge the lines and plot
combined_lines = lines[0].merge(lines[1:])
combined_lines.plot(line_width=5, cpos='xy', background='w', color='k')

# note: the above does not convert to PolyData on pyvista<0.32.0

# as_ribbons = combined_lines.ribbon(width=1E-5, normal=(0, 0, 1))
# as_ribbons.plot(cpos='xy')
```
![tmp](https://user-images.githubusercontent.com/11981631/130704092-85a0e35f-f8fb-4e28-b175-698f4783e382.png)

The only catch is you'll need to either wait for our upcoming release or install our development (main) branch.

**Comment by josh-millar** [comment]
Ah thanks @akaszynski, looks great, will wait for the release. Did you manage to find the source of the weird behaviour in the multiblock?

**Comment by akaszynski** [comment]
> Did you manage to find the source of the weird behaviour in the multiblock?

Yes, it has to do with how these lines are converted to an unstructured grid.  I'd avoid using MultiBlock for this reason and just merge PolyData.
